### DISCUSSION TITLE
Toggle to Welsh

### DISCUSSION THREAD

**Discussion by michaelcattell** [dwp/design-system#450] [Q&A]
Hello, we'd like to use this open discussion on the use of the **Toggle to Welsh**:

Some discussion points to get things going:
Have you used or know where this component has been used?
Do you know if there's been any research or testing done on it or around it?
Do you agree with the user need?
Things we don’t know enough?

Link to backlog: https://github.com/dwp/design-system-community-backlog/issues/26
Link to Welsh Toggle within UC: https://dwpdigital.blog.gov.uk/2018/12/18/credyd-cynhwysol-universal-credit-in-welsh/

**Comment by martinwake** [comment]
I showed this to the DWP content design community on 26 November 2021. Questions:

- Should the toggle be on the start page AND on every page of the service? The guidance at the moment suggests it is only needed on the start page if it can't be put on every page.

**Comment by KristinaIankova** [comment]
Hello from MOJ,

We just had an accessibility audit which included this component. The following was flagged.

```

The Welsh language toggle link, labeled “Cymraeg,” is intended to allow users to switch the language of the interface to Welsh. However, screen readers like NVDA, JAWS, and VoiceOver do not pronounce the Welsh text correctly.

Although the element containing the aria-label “Newid iaith i’r Gymraeg” has the language attribute correctly set to Welsh (lang="cy"), screen readers are unable to provide accurate Welsh pronunciation as they do not have full support switching to the Welsh Language. As a result, these screen readers pronounce the text phonetically based on English pronunciation rules, leading to an unintelligible output.

This issue affects screen reader users who rely on accurate pronunciation to understand the content and navigation options. Mispronunciation of language-specific words can lead to confusion and hinder the user experience.

Note that this is not an issue on TalkBack; the phrase is announced in Welsh automatically as expected.
```

Their suggested solution was removing the aria-label since it overrides the displayed text ‘Cymraeg’. Although screen readers will still announce this as “sim reeg”, this would be easier to interpret than “Newid iaith i’r Gymraeg”.

Our preferred solution is to change the aria label to english since the user will be on the english page at this point.

Any thoughts?
